From 6c431dc9ddaf16d9b743fef42623beec9700d65d Mon Sep 17 00:00:00 2001
From: Uli Schlachter <psychon@znc.in>
Date: Tue, 9 Mar 2021 07:53:50 +0100
Subject: [PATCH 07/11] Avoid a use-after-scope

This is the same fix as commit b345be5afee, but in a different place in
the same file.

Fixes: https://gitlab.freedesktop.org/cairo/cairo/-/issues/453
Signed-off-by: Uli Schlachter <psychon@znc.in>
---
 src/cairo-spans-compositor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cairo-spans-compositor.c b/src/cairo-spans-compositor.c
index 5f956ca98..50c92b25c 100644
--- a/src/cairo-spans-compositor.c
+++ b/src/cairo-spans-compositor.c
@@ -1041,6 +1041,7 @@ _cairo_spans_compositor_stroke (const cairo_compositor_t	*_compositor,
 
     if (status == CAIRO_INT_STATUS_UNSUPPORTED) {
 	cairo_polygon_t polygon;
+	cairo_box_t limits;
 	cairo_fill_rule_t fill_rule = CAIRO_FILL_RULE_WINDING;
 
 	if (! _cairo_rectangle_contains_rectangle (&extents->unbounded,
@@ -1049,7 +1050,6 @@ _cairo_spans_compositor_stroke (const cairo_compositor_t	*_compositor,
 	    if (extents->clip->num_boxes == 1) {
 		_cairo_polygon_init (&polygon, extents->clip->boxes, 1);
 	    } else {
-		cairo_box_t limits;
 		_cairo_box_from_rectangle (&limits, &extents->unbounded);
 		_cairo_polygon_init (&polygon, &limits, 1);
 	    }
-- 
2.33.0.windows.2

